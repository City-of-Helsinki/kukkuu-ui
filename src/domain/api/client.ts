import ApolloClient from 'apollo-boost';

import { store } from '../app/state/AppStore';
import { apiTokenSelector } from '../auth/state/AuthenticationSelectors';

export default new ApolloClient({
  request: async operation => {
    try {
      const token = apiTokenSelector(store.getState());
      operation.setContext({
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
    } catch (e) {
      // User not authenticated
      // eslint-disable-next-line no-console
      console.error(e);
      // TODO: add error-handler
    }
  },
  uri: process.env.REACT_APP_API_URI,
});
