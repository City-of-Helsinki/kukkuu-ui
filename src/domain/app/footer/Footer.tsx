import { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import joinClassNames from 'classnames';
import { Koros } from 'hds-react';

import styles from './footer.module.scss';
import Container from '../layout/Container';
import { getCurrentLanguage } from '../../../common/translation/TranslationUtils';

const Footer: FunctionComponent = () => {
  const { t, i18n } = useTranslation();
  const currentLocale = getCurrentLanguage(i18n);
  const logoLang = currentLocale === 'sv' ? styles.sv : styles.fi;

  return (
    <div className={styles.footerWrapper}>
      <Koros className={styles.koros} />
      <Container>
        <div className={styles.footer}>
          <div className={joinClassNames(styles.helsinkiLogo, logoLang)}></div>
          <div className={styles.copyright}>
            <p>{t('footer.copyrightText')}</p>
          </div>
          <div className={styles.links}>
            <Link to="/accessibility#start">
              {t('accessibilityStatement.title')}
            </Link>{' '}
            • <Link to="/terms#">{t('termsOfService.title')}</Link> •{' '}
            <a
              href={t('descriptionOfTheFile.url')}
              target="_blank"
              rel="noopener noreferrer"
            >
              {t('descriptionOfTheFile.title')}
            </a>{' '}
            •{' '}
            <a href={t('dataProtection.url')} rel="noopener noreferrer">
              {t('dataProtection.title')}
            </a>
          </div>
        </div>
      </Container>
    </div>
  );
};

export default Footer;
