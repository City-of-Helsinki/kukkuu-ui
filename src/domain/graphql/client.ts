import ApolloClient from 'apollo-boost';

import getAuthenticatedUser from '../auth/getAuthenticatedUser';

export default new ApolloClient({
  request: async operation => {
    try {
      const user = await getAuthenticatedUser();
      operation.setContext({
        headers: {
          Authorization: `Bearer ${user.id_token}`,
        },
      });
    } catch (e) {
      // User not authenticated
      console.error(e);
    }
  },
  uri: process.env.REACT_APP_PROFILE_GRAPHQL,
});
