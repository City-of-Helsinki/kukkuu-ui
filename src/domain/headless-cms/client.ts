import {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
  ApolloLink,
} from '@apollo/client';
import { onError } from '@apollo/client/link/error';
import * as Sentry from '@sentry/browser';

const httpLink = createHttpLink({
  uri: process.env.REACT_APP_CMS_URI,
});

const errorLink = onError(({ graphQLErrors, networkError }) => {
  if (graphQLErrors) {
    graphQLErrors.forEach(({ message, locations, path, extensions }) => {
      const errorCode = extensions?.code;

      // eslint-disable-next-line max-len
      const errorMessage = `[GraphQL error]: Message: ${message}, Code: ${errorCode}, Location: ${locations}, Path: ${path}`;

      Sentry.captureMessage(errorMessage);

      if (process.env.NODE_ENV === 'development') {
        // eslint-disable-next-line no-console
        console.error(errorMessage);
      }
    });
  }

  if (networkError) {
    Sentry.captureMessage('Network error');
  }
});

const client = new ApolloClient({
  link: ApolloLink.from([errorLink, httpLink]),
  cache: new InMemoryCache(),
});

export default client;
