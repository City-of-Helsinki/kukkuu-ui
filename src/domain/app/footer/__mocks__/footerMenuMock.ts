/* eslint-disable max-len */
import { MenuDocument } from 'react-helsinki-headless-cms/apollo';
import { MockedResponse } from '@apollo/client/testing';

import { languageToMenuNameMap } from '../constants';

/**
 * This is a mock of the response from the footer menu query.
 * Data taken from production CMS query response.
 */
export const footerMenuQueryResponse = {
  data: {
    menu: {
      id: 'dGVybTo1Nw==',
      menuItems: {
        nodes: [
          {
            id: 'cG9zdDo0MjY=',
            parentId: null,
            order: 1,
            target: null,
            title: null,
            path: 'https://kukku.fi/fi',
            label: 'Kulttuurin kummilapset',
            connectedNode: null,
            __typename: 'MenuItem',
          },
          {
            id: 'cG9zdDo0Mjc=',
            parentId: null,
            order: 2,
            target: null,
            title: null,
            path: 'https://hkih.production.geniem.io/saavutettavuusseloste',
            label: 'Saavutettavuusseloste',
            connectedNode: null,
            __typename: 'MenuItem',
          },
        ],
        __typename: 'MenuToMenuItemConnection',
      },
      __typename: 'Menu',
    },
  },
} as const;

export const footerMenuMock: MockedResponse = {
  request: {
    query: MenuDocument,
    variables: {
      id: languageToMenuNameMap['fi'],
      menuIdentifiersOnly: true,
    },
  },
  result: footerMenuQueryResponse,
};

export const emptyFooterMenuQueryResponse = {
  data: {
    menu: {
      id: 'empty-footer-menu',
      menuItems: {
        nodes: [],
      },
    },
  },
} as const;
