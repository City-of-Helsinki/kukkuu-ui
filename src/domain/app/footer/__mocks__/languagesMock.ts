import { LanguagesDocument } from 'react-helsinki-headless-cms/apollo';
import { MockedResponse } from '@apollo/client/testing';

import { HARDCODED_CMS_LANGUAGE_QUERY_RESPONSE } from '../../../languages/constants';

export const languagesMock: MockedResponse = {
  request: {
    query: LanguagesDocument,
  },
  result: { ...HARDCODED_CMS_LANGUAGE_QUERY_RESPONSE },
};
