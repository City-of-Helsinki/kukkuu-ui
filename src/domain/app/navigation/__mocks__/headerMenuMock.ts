import { MenuDocument } from 'react-helsinki-headless-cms/apollo';
import { MockedResponse } from '@apollo/client/testing';

import { languageToMenuNameMap } from '../Navigation';

/**
 * This is a mock of the response from the header menu query.
 * Data taken from production CMS query response, and strongly
 * reduced in size for simplicity.
 */
export const headerMenuQueryResponse = {
  data: {
    menu: {
      id: 'dGVybToxNg==',
      menuItems: {
        nodes: [
          {
            id: 'cG9zdDo1MTU=',
            parentId: null,
            order: 1,
            target: null,
            title: null,
            path: '/ajankohtaista/',
            label: 'Ajankohtaista',
            connectedNode: {
              node: {
                id: 'cG9zdDo1MDU=',
                content: '<p>Kirjautuminen palveluun on muuttunut</p>',
                slug: 'ajankohtaista',
                title: 'Ajankohtaista',
                uri: '/ajankohtaista/',
                link: 'https://kukkuu.content.api.hel.fi/ajankohtaista/',
                lead: '',
                seo: {
                  title: 'Ajankohtaista &#x2d; Kukkuu',
                  description: 'Kirjautuminen palveluun on muuttunut',
                  openGraphTitle: 'Ajankohtaista',
                  openGraphDescription: 'Kirjautuminen palveluun on muuttunut',
                  openGraphType: 'website',
                  twitterTitle: 'Ajankohtaista',
                  twitterDescription: 'Kirjautuminen palveluun on muuttunut',
                  canonicalUrl: '',
                  socialImage: null,
                  __typename: 'SEO',
                },
                hero: {
                  background_color: '',
                  background_image_url: '',
                  description: '',
                  link: {
                    target: null,
                    title: null,
                    url: null,
                    __typename: 'Link',
                  },
                  title: '',
                  wave_motif: '',
                  __typename: 'Hero',
                },
                language: {
                  code: 'FI',
                  id: 'TGFuZ3VhZ2U6Zmk=',
                  locale: 'fi',
                  name: 'Suomi',
                  slug: 'fi',
                  __typename: 'Language',
                },
                translations: [],
                featuredImage: null,
                breadcrumbs: [
                  {
                    title: 'Etusivu',
                    uri: '/',
                    __typename: 'Breadcrumb',
                  },
                  {
                    title: 'Ajankohtaista',
                    uri: '/ajankohtaista/',
                    __typename: 'Breadcrumb',
                  },
                ],
                sidebar: [],
                modules: [],
                __typename: 'Page',
                children: {
                  nodes: [],
                  __typename:
                    'HierarchicalContentNodeToContentNodeChildrenConnection',
                },
              },
              __typename: 'MenuItemToMenuItemLinkableConnectionEdge',
            },
            __typename: 'MenuItem',
          },
        ],
        __typename: 'MenuToMenuItemConnection',
      },
      __typename: 'Menu',
    },
  },
} as const;

export const headerMenuMock: MockedResponse = {
  request: {
    query: MenuDocument,
    variables: {
      id: languageToMenuNameMap['fi'],
      menuIdentifiersOnly: false,
    },
  },
  result: headerMenuQueryResponse,
};

export const emptyHeaderMenuQueryResponse = {
  data: {
    menu: {
      id: 'empty-header-menu',
      menuItems: {
        nodes: [],
      },
    },
  },
} as const;
