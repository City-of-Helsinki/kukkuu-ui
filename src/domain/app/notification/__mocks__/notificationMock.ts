import { MockedResponse } from '@apollo/client/testing';

import notificationQuery from '../notificationQuery';

/**
 * This is a mock of the response from the notification query.
 * Data taken from production CMS query response, and cleaned/simplified.
 */
export const notificationQueryResponse = {
  data: {
    notification: {
      content:
        '<p>Kirjautuminen Kulttuurin kummilapset -palveluun on muuttunut 17.6.2024 alkaen.</p>',
      title: 'Huom.!',
      level: 'info',
      startDate: '2024-06-17T00:00:00+03:00',
      endDate: '',
      linkText: 'Lue lisää',
      linkUrl: 'https://kummilapset.hel.fi/fi/ajankohtaista/',
      __typename: 'Notification',
    },
  },
} as const;

export const notificationMock: MockedResponse = {
  request: {
    query: notificationQuery,
    variables: { language: 'fi' },
  },
  result: notificationQueryResponse,
};
