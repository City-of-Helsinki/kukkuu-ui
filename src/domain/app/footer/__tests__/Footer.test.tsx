import * as React from 'react';
import { MockedResponse } from '@apollo/client/testing';
import { vi } from 'vitest';

import Footer from '../Footer';
import { render, screen } from '../../../../common/test/testingLibraryUtils';
import { footerMenuMock } from '../__mocks__/footerMenuMock';
import { languagesMock } from '../__mocks__/languagesMock';

vi.mock('hds-react', async () => {
  const actual = await vi.importActual('hds-react');
  return {
    ...actual,
    logoFi: 'mocked hds-react logoFi',
  };
});

const mocks: MockedResponse[] = [{ ...footerMenuMock }, { ...languagesMock }];

it('Footer matches snapshot', async () => {
  const { container } = render(<Footer />, mocks);
  await screen.findByText('Saavutettavuusseloste');
  expect(container.firstChild).toMatchSnapshot();
});
