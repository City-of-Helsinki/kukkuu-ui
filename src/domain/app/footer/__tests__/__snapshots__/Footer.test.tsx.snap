// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `"<footer class=\\"footerWrapper\\"><div class=\\"container\\"><div class=\\"footer\\"><div class=\\"helsinkiLogo\\"></div><div class=\\"copyright\\"><p>© Copyright 2020 • Kaikki oikeudet pidätetään</p></div><div class=\\"links\\"><a href=\\"/accessibility#start\\">Saavutettavuusseloste</a> • <a href=\\"/terms\\">Käyttöehdot</a> • <a href=\\"https://www.hel.fi/static/liitteet/kanslia/rekisteriselosteet/Kuva/Kuva-EU-Kaupunginorkesterin-kummiperherekisteri.pdf\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">Rekisteriseloste</a></div></div></div></footer>"`;
